src/main/java/org/graylog/aws/kinesis/KinesisConsumer.java

```diff
@@ -48,37 +55,67 @@
     private final HttpUrl proxyUrl;
     private final AWSPluginConfiguration awsConfig;
     private final Consumer<byte[]> dataHandler;
+    private final Integer maxThrottledWaitMillis;
+    private final Integer recordBatchSize;
 
     private Worker worker;
+    private KinesisTransport transport;
+    private final ObjectMapper objectMapper;
+    /**
+     * Checkpointing must be performed when the KinesisConsumer needs to be shuts down due to sustained throttling.
+     * At the time when shutdown occurs, checkpointing might not have happened for a while, so we keep track of the
+     * last sequence to checkpoint to. */
+    private String lastSuccessfulRecordSequence = null;
 
     public KinesisConsumer(String kinesisStreamName,
                            Region region,
                            Consumer<byte[]> dataHandler,
                            AWSPluginConfiguration awsConfig,
                            AWSAuthProvider authProvider,
                            NodeId nodeId,
-                           @Nullable HttpUrl proxyUrl) {
+                           @Nullable HttpUrl proxyUrl,
+                           KinesisTransport transport,
+                           ObjectMapper objectMapper,
+                           Integer maxThrottledWaitMillis,
+                           Integer recordBatchSize) {
         this.kinesisStreamName = requireNonNull(kinesisStreamName, "kinesisStreamName");
         this.region = requireNonNull(region, "region");
         this.dataHandler = requireNonNull(dataHandler, "dataHandler");
         this.awsConfig = requireNonNull(awsConfig, "awsConfig");
         this.authProvider = requireNonNull(authProvider, "authProvider");
         this.nodeId = requireNonNull(nodeId, "nodeId");
         this.proxyUrl = proxyUrl;
+        this.transport = transport;
+        this.objectMapper = objectMapper;
+        this.recordBatchSize = recordBatchSize;
+
+        // Use default throttled time of 60 seconds if not specified in configuration.
+        this.maxThrottledWaitMillis =
+                maxThrottledWaitMillis != null ? maxThrottledWaitMillis : KinesisTransport.DEFAULT_THROTTLED_WAIT;
```

Comment:
Move this default to second argument in KinesisTransport constructor (config read)